app/global/MainModule.java

```diff
@@ -804,4 +814,58 @@ public ActorRef get() {
         private final Config _configuration;
         private final boolean _enabled;
     }
+
+    private static final class RollupConsistencyCheckerQueueProvider implements Provider<ActorRef> {
```

Comment:
Doesn't this need to be either a cluster singleton or a sharded actor?

Reply:
I am no Akka expert, but I thought that, as written, this would result in each host having its own queue, independent of the other hosts' queues. (And its own consistency-checkers, independent of the other hosts' checkers.) Yeah, this will result in unequal load distribution, but that seems fine, given how little (of MP's) resources the consistency-checking operation consumes.

Reply:
Okay, just wanted to make sure that was a conscious decision.

Reply:
I am also curious about this from the API point of view.

The controller allows users to submit tasks, but there is no way to track them or know their result because each host is independent. What's the plan here going forward? (e.g. what does this on-demand/user-initiated use case going to look like?)

^ Not saying it has to be addressed now, just want to understand what that plan is and how independent actors per host fits into it.

Reply:
I currently don't intend for there to _be_ a user experience; I only added an API endpoint for debugging purposes.

At some point in the future, we might decide to give the consistency-checker a real UI so that people can submit tasks without going through us; and when that happens, we can add persistence or clustering or whatever. And maybe that switchover will cause some transient issues and result in some tasks getting dropped, but that's okay, because only _after_ that point do we need to start offering reliability guarantees. Until then, it's A-okay for deployments to randomly drop some tasks: the consistency-checker doesn't need to be resilient to occasional transient disruptions.

Tldr: whatever features we add to the consistency-checker in the future, its core is _basically_ going to look like what this diff lays out: some pool of tasks that executors pull from. And because it doesn't need any particularly strong reliability guarantees, it's unusually easy to change.

Reply:
Okay, please document this clearly.